pkg/policyendpoints/manager.go

```diff
@@ -348,3 +239,203 @@ func (m *policyEndpointsManager) getEndpointInfoKey(info policyinfo.EndpointInfo
+	if chunkEndIdx < len(ingressList) {
+		ingressRuleChunks := lo.Chunk(ingressList[chunkEndIdx:], m.endpointChunkSize)
+		for _, chunk := range ingressRuleChunks {
+			newEP := m.newPolicyEndpoint(policy, m.getListOfEndpointInfoFromHash(chunk, rulesMap), nil, nil)
+			createPolicyEndpoints = append(createPolicyEndpoints, newEP)
+		}
+	}
+	return createPolicyEndpoints, doNotDelete
+}
+
+// packingEgressRules iterates over egress rules across available policy endpoints and required egress rule changes.
+// it returns the egress rules packed in policy endpoints and a set of policy endpoints that need to be kept.
+func (m *policyEndpointsManager) packingEgressRules(policy *networking.NetworkPolicy,
+	rulesMap map[string]policyinfo.EndpointInfo,
+	createPolicyEndpoints, modifiedEndpoints, potentialDeletes []policyinfo.PolicyEndpoint) ([]policyinfo.PolicyEndpoint, sets.Set[types.NamespacedName]) {
+	// ingressRuleChunks := lo.Chunk(maps.Keys(ingressEndpointsMap), m.endpointChunkSize)
+	doNotDelete := sets.Set[types.NamespacedName]{}
+	chunkStartIdx := 0
+	chunkEndIdx := 0
+	egressList := maps.Keys(rulesMap)
+	for _, sliceToCheck := range [][]policyinfo.PolicyEndpoint{createPolicyEndpoints, modifiedEndpoints, potentialDeletes} {
+		for i := range sliceToCheck {
+			// reset start pointer if end pointer is updated
+			chunkStartIdx = chunkEndIdx
+			// Instead of adding the entire chunk we should try to add to full the slice
+			if len(sliceToCheck[i].Spec.Egress) < m.endpointChunkSize && chunkEndIdx < len(egressList) {
+				for len(sliceToCheck[i].Spec.Egress)+(chunkEndIdx-chunkStartIdx+1) < m.endpointChunkSize && chunkEndIdx < len(egressList)-1 {
+					chunkEndIdx++
+				}
+
+				sliceToCheck[i].Spec.Egress = append(sliceToCheck[i].Spec.Egress, m.getListOfEndpointInfoFromHash(lo.Slice(egressList, chunkStartIdx, chunkEndIdx+1), rulesMap)...)
+				// move the end to next available index to prepare next appending
+				chunkEndIdx++
+			}
+			// as long as the second pointer moves, we need to include the PE
+			if chunkStartIdx != chunkEndIdx {
+				doNotDelete.Insert(k8s.NamespacedName(&sliceToCheck[i]))
+			}
+		}
+	}
+
+	// if the incoming ingress rules haven't been all processed yet, we need new PE(s).
+	if chunkEndIdx < len(egressList) {
+		egressRuleChunks := lo.Chunk(egressList[chunkEndIdx:], m.endpointChunkSize)
+		for _, chunk := range egressRuleChunks {
+			newEP := m.newPolicyEndpoint(policy, nil, m.getListOfEndpointInfoFromHash(chunk, rulesMap), nil)
+			createPolicyEndpoints = append(createPolicyEndpoints, newEP)
+		}
+	}
+	return createPolicyEndpoints, doNotDelete
+}
+
+// packingPodSelectors iterates over pod selectors across available policy endpoints and required pod selector changes.
+// it returns the pod selectors packed in policy endpoints and a set of policy endpoints that need to be kept.
+func (m *policyEndpointsManager) packingPodSelectors(policy *networking.NetworkPolicy,
```

Comment:
nit: `packingPodSelectors` -> `packingPodSelectorEndpoints`

Reply:
Updated the name. Thanks